### PR TITLE
[lzokay] update to the latest for newer compilers

### PR DESCRIPTION
**Pull request by n-taka**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

It's my first time to make a PR to vcpkg. If you find any mistakes, please let me know!

### CODE REVIEW

**Comment by n-taka** [comment]
@microsoft-github-policy-service agree company="YOSHIMOTO ART FACTORY LLC"

File: ports/lzokay/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "lzokay",
   "version-date": "2020-07-30",
-  "port-version": 2,
+  "port-version": 3,
   "description": "lzokay – MIT licensed C++ implementation of LZO compression/decompression algorithm",
   "homepage": "https://github.com/jackoalan/lzokay",
```
Comment (WangWeiLin-MV) [inline ports/lzokay/vcpkg.json:6]:
```suggestion
  "homepage": "https://github.com/AxioDL/lzokay",
```

File: ports/lzokay/vcpkg.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "lzokay",
   "version-date": "2020-07-30",
-  "port-version": 2,
+  "port-version": 3,
```
Comment (WangWeiLin-MV) [inline ports/lzokay/vcpkg.json:4]: How about update the `version-date` instead of the `port-version`?

Comment (n-taka) [reply ports/lzokay/vcpkg.json:4]: Previous versions (namely, port-version `1` and `2`) updated `port-version` (not `version-date`) for some reason.
I'm a newbie about versioning in vcpkg, and don't know what happens when I update `version-date`...

Comment (WangWeiLin-MV) [reply ports/lzokay/vcpkg.json:4]: Could be changed as

```diff
-    "version-date": "2020-07-30",
-    "port-version": 2,
+    "version-date": "2023-10-22",
```

Comment (WangWeiLin-MV) [reply ports/lzokay/vcpkg.json:4]: And the re-run `./vcpkg x-add-version --all` to update the version json.

Comment (n-taka) [reply ports/lzokay/vcpkg.json:4]: Thank you! I try this.

File: versions/l-/lzokay.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "489181d9c70f9d80f622ce33000a652329d723dc",
+      "version-date": "2023-10-22",
+      "port-version": 0
+    },
+    {
+      "git-tree": "8d04e6eee122ebca45dc5b9dfc0f1aed4357d310",
+      "version-date": "2020-07-30",
+      "port-version": 3
+    },
```
Comment (WangWeiLin-MV) [inline versions/l-/lzokay.json:12]:
```suggestion
```

**Review by WangWeiLin-MV — APPROVED** [review]
The port installation tests pass with the following triplets:

* x64-windows

**Pull request merged**